src/crimson/os/seastore/lba_manager/btree/lba_btree_node.h

```diff
@@ -136,72 +136,157 @@ struct lba_map_val_le_t {
   }
 };
 
+template <bool val>
+struct leaf_node_type_t {};
+
+template<>
+struct leaf_node_type_t<true> {
+  static constexpr extent_types_t type = extent_types_t::LADDR_LEAF;
+};
+
+template<>
+struct leaf_node_type_t<false> {
+  static constexpr extent_types_t type = extent_types_t::DINK_LADDR_LEAF;
+};
+
+template <bool has_children>
```

Comment:
This is because leaf nodes in LBA unit tests don't have child nodes and a lot of assertion checks in the code don't hold. So I added this `DINK_LADDR_LEAF` type to distinguish this situation from normal cases. `DINK` means parents that are not going to have children.

The reason why I used templates for this is that, if not templates, probably I need to add a field in LBALeafNode to do the distinguishing or use a macro for this, the former would cause cpu overhead when running in normal cases and the latter may lead to a coding style pretty like the "template" way.

What do you think?

Reply:
see https://github.com/ceph/ceph/pull/47749#discussion_r1125983768